app/controllers/facebooks_controller.rb

```diff
@@ -0,0 +1,38 @@
+class FacebooksController < ApplicationController
+  require 'net/http'
+  
+  def login_facebook
+    response_body = generic_callback
```

Comment:
giải thích cho anh từng đoạn của method login_facebook làm những gì nhé =))

Reply:
vâng cho em it phut ạ

Reply:
method generic_callback nó sẽ gửi lên phía facebook một request trong url có 4 tham số:
- FACEBOOK_APP_ID
- FACEBOOK_APP_SECRET
- param[:code] cái này nhận được khi user click vào button sign in with facebbook sau đó đồng  ý với permission sau đó redirect facebook controller -> action login facebook
-FACEBOOK_APP_URL cái này dang https/ http vì facese báo đương dẫn không an toàn nếu dùng localhost
cái chúng ta nhận được là access_token mà facebook trả lại 
- chúng ta  dùng cái access_token để lấy thông tin của user(email, name)
-  em tìm trong DB thông qua email nêu có thì update lại access_token, còn chưa có thì tạo mới